app/src/main/java/net/gsantner/markor/ui/NewFileDialog.java

```diff
@@ -115,6 +130,20 @@ public void onNothingSelected(AdapterView<?> adapterView) {
             }
         });
 
+        encryptCheckbox.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            final String currentExtention = fileExtEdit.getText().toString();
+            if (isChecked) {
+                if (!currentExtention.endsWith(Constants.ENCRYPTION_EXTENSION)) {
+                    fileExtEdit.setText(currentExtention + Constants.ENCRYPTION_EXTENSION);
+                }
+            } else {
+                if (currentExtention.endsWith(Constants.ENCRYPTION_EXTENSION)) {
+                    fileExtEdit.setText(currentExtention.substring(0, currentExtention.length() - Constants.ENCRYPTION_EXTENSION.length()));
```

Comment:
please try what happens if existing extension is empty. potential outofrange error

Reply:
Tested the code is safe.